File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int  n = prices.size();
        bool stock = false;
        int  profit = 0;
        for(int i=0;i<n-1;i++){
            if(!stock && prices[i+1]>prices[i]) {
                profit-=prices[i];
                stock = true;
            }
            else if(stock && prices[i+1]<prices[i]){
                profit+=prices[i];
                stock = false;
            }
        }
        if(stock) profit+=prices[n-1];
        return profit;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minV = float('inf')
        profit  = 0;
        
        for i in range(len(prices)):
            if prices[i]<minV:
                minV = prices[i]
            profit  = max(profit, prices[i]-minV)
        return profit